### ISSUE TITLE
Use a context manager

### ISSUE THREAD

**Issue by stevepiercy** [Pylons/pyramid-cookiecutter-zodb#21]
refs: https://github.com/Pylons/pyramid/issues/3069

**Comment by stevepiercy** [comment]
Well, I tried the code snippet below in `__init__.py` and dozens of variations, none of which succeeded. I would always get:

```text
404 Not Found

The resource could not be found.

/
``` 

I just don't know enough about context managers in Python to get this to work. I noticed only one pattern in almost every code sample in the Pyramid docs where `if __name__ == '__main__':` is used whereas in the cookiecutters `def main` is used.

```python
def main(global_config, **settings):
    """ This function returns a Pyramid WSGI application.
    """
    config = Configurator(root_factory=root_factory, settings=settings)
    settings = config.get_settings()
    settings['tm.manager_hook'] = 'pyramid_tm.explicit_manager'
    with Configurator() as config:
        config.include('pyramid_chameleon')
        config.include('pyramid_tm')
        config.include('pyramid_retry')
        config.include('pyramid_zodbconn')
        config.add_static_view('static', 'static', cache_max_age=3600)
        config.scan()
        return config.make_wsgi_app()
```

Hopefully someone can help out, as I've run out of ideas.

**Comment by mmerickel** [comment]
How many configurators do you need?

**Comment by stevepiercy** [comment]
oh. derp. Just the one, sensei. :grin:

**Comment by stevepiercy** [comment]
Closed by PRs #26, #27, #28
